deepmd/pt/utils/stat.py

```
@@ -35,51 +35,190 @@
+                collect_values = torch.unique(element_data.flatten(), sorted=True)
+                for elem in collect_values.tolist():
+                    frames_with_elem = torch.any(element_data == elem, dim=1)
+                    row_indices = torch.where(frames_with_elem)[0]
+                    collect_ele[elem] += len(row_indices)
+        finalize_stats(sys_stat)
         lst.append(sys_stat)
+
+        # get frame index
+        if datasets[0].mixed_type and enable_element_completion:
+            element_counts, type_map = dataset.get_frame_index_for_elements()
+            for new_idx, elem_name in type_name.items():
```

Ruff (0.8.2)

Undefined name `type_name`

(F821)